src/validator.ts

```diff
@@ -1,22 +1,61 @@
 import {
   FieldValidationFunctionSync,
   parseMessageWithCustomArgs,
+  ValidationResult,
 } from '@lemoncode/fonk';
 
-// TODO: Add validator type
-const VALIDATOR_TYPE = '';
+export const VALIDATOR_TYPE = 'BIC_VALIDATOR';
 
-// TODO: Add default message
-let defaultMessage = '';
+let defaultMessage = 'This is not in BIC correct format';
 export const setErrorMessage = message => (defaultMessage = message);
 
-const isDefined = value => value !== void 0 && value !== null && value !== '';
+const isDefined = (value: string) =>
+  value !== void 0 && value !== null && value !== '';
 
-export const validator: FieldValidationFunctionSync = fieldValidatorArgs => {
+export const validator: FieldValidationFunctionSync = (
+  fieldValidatorArgs
+): ValidationResult => {
   const { value, message = defaultMessage, customArgs } = fieldValidatorArgs;
 
-  // TODO: Add validator
-  const succeeded = !isDefined(value) || ...;
+  const matchAlfhanumericExpresion = (value: string): boolean => {
+    const letterAndNumbers = /^[0-9a-zA-Z]+$/;
+    return value.match(letterAndNumbers) ? true : false;
+  };
+  const hasCorrectLength = (value: string): boolean => {
+    return value.length === 8 || value.length === 11;
+  };
+  const validFormatBusinessPartyIdentifier = (
+    value: string,
+    prefix: boolean
+  ): boolean => {
+    if (prefix) {
+      value = value.substring(0, 4);
+    } else {
+      value = value.substring(6, 8);
+    }
+    return matchAlfhanumericExpresion(value);
+  };
+  const validFormatCountryCode = (value: string): boolean => {
+    const letters = /^[A-Za-z]+$/;
+    value = value.substring(4, 6);
+    return value.match(letters) ? true : false;
+  };
+  const validateOptionalPart = (value: string): boolean => {
+    if (value.length > 8) {
+      value = value.substring(8, 11);
+      return matchAlfhanumericExpresion(value);
+    } else {
+      return true;
+    }
+  };
+
+  const succeeded =
+    !isDefined(value) ||
+    (hasCorrectLength(value) &&
+      validFormatBusinessPartyIdentifier(value, true) &&
+      validFormatCountryCode(value) &&
+      validFormatBusinessPartyIdentifier(value, false) &&
+      validateOptionalPart(value));
```

Comment:
Add a check for the ISO country code: 

https://www.upct.es/relaciones_internacionales/prog/docs/Erasmus-14-15/iniciales_paises_iso.pdf

Take care with Kosovo, has an own format

https://en.wikipedia.org/wiki/ISO_9362

Reply:
I'm in doubt here, if I made a regular expression for the BIC code format entirely, why I need to check the country code apart?